spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/DialectResolver.java

```diff
@@ -103,79 +79,18 @@ public interface JdbcDialectProvider {
-
-		@Override
-		public Optional<Dialect> getDialect(JdbcOperations operations) {
-			return Optional.ofNullable(operations.execute((ConnectionCallback<Dialect>) DefaultDialectProvider::getDialect));
-		}
-
-		@Nullable
-		private static JdbcDialect getDialect(Connection connection) throws SQLException {
-
-			DatabaseMetaData metaData = connection.getMetaData();
-
-			String name = metaData.getDatabaseProductName().toLowerCase(Locale.ENGLISH);
+	@Deprecated(since = "3.5", forRemoval = true)
+	static public class DefaultDialectProvider extends
+			org.springframework.data.jdbc.core.dialect.DialectResolver.DefaultDialectProvider implements JdbcDialectProvider {
 
-			if (name.contains("hsql")) {
-				return JdbcHsqlDbDialect.INSTANCE;
-			}
-			if (name.contains("h2")) {
-				return JdbcH2Dialect.INSTANCE;
-			}
-			if (name.contains("mysql")) {
-				return new JdbcMySqlDialect(getIdentifierProcessing(metaData));
-			}
-			if (name.contains("mariadb")) {
-				return new JdbcMariaDbDialect(getIdentifierProcessing(metaData));
-			}
-			if (name.contains("postgresql")) {
-				return JdbcPostgresDialect.INSTANCE;
-			}
-			if (name.contains("microsoft")) {
-				return JdbcSqlServerDialect.INSTANCE;
-			}
-			if (name.contains("db2")) {
-				return JdbcDb2Dialect.INSTANCE;
-			}
-			if (name.contains("oracle")) {
-				return JdbcOracleDialect.INSTANCE;
-			}
-
-			LOG.info(String.format("Couldn't determine Dialect for \"%s\"", name));
-			return null;
-		}
-
-		private static IdentifierProcessing getIdentifierProcessing(DatabaseMetaData metaData) throws SQLException {
-
-			// getIdentifierQuoteString() returns a space " " if identifier quoting is not
-			// supported.
-			String quoteString = metaData.getIdentifierQuoteString();
-			IdentifierProcessing.Quoting quoting = StringUtils.hasText(quoteString)
-					? new IdentifierProcessing.Quoting(quoteString)
-					: IdentifierProcessing.Quoting.NONE;
-
-			IdentifierProcessing.LetterCasing letterCasing;
-			// IdentifierProcessing tries to mimic the behavior of unquoted identifiers for their quoted variants.
-			if (metaData.supportsMixedCaseIdentifiers()) {
-				letterCasing = IdentifierProcessing.LetterCasing.AS_IS;
-			} else if (metaData.storesUpperCaseIdentifiers()) {
-				letterCasing = IdentifierProcessing.LetterCasing.UPPER_CASE;
-			} else if (metaData.storesLowerCaseIdentifiers()) {
-				letterCasing = IdentifierProcessing.LetterCasing.LOWER_CASE;
-			} else { // this shouldn't happen since one of the previous cases should be true.
-				// But if it does happen, we go with the ANSI default.
-				letterCasing = IdentifierProcessing.LetterCasing.UPPER_CASE;
-			}
-
-			return IdentifierProcessing.create(quoting, letterCasing);
-		}
 	}
 
 	/**
 	 * Exception thrown when {@link DialectResolver} cannot resolve a {@link Dialect}.
 	 */
-	public static class NoDialectException extends NonTransientDataAccessException {
+	@Deprecated(since = "3.5", forRemoval = true)
```

Comment:
Maybe we can add a javadoc `@deprecated` tag here? Just in case to specify the replacement like in other parts of the old `DialectResolver`...